lib/src/handle_access_token_response.dart

```diff
@@ -17,6 +17,54 @@ import 'authorization_exception.dart';
 /// amount of time.
 const _expirationGrace = const Duration(seconds: 10);
 
+/// Returns a function that that parses parameters from a response with either a JSON or URL-encoded body.
+Function parseJsonOrUrlEncoded(void validate(bool condition, String message)) {
+  return (String contentTypeString, String body) {
+    var contentType = contentTypeString == null
+        ? null
+        : new MediaType.parse(contentTypeString);
+
+    // The spec requires a content-type of application/json, but some endpoints
+    // (e.g. Dropbox) serve it as text/javascript instead.
+    validate(contentType != null &&
```

Comment:
Passing in `validate()` seems like a code smell to me. I think the `getParameters()` function should just throw a `FormatException` with a message, and `handleAccessTokenResponse()` should catch that exception and wrap it with extra information about the endpoint and response body.